testsuite/MDAnalysisTests/analysis/test_psa.py

```diff
@@ -134,6 +134,54 @@ def test_reversal_frechet(self, frech_matrix):
         err_msg = "Frechet distances did not increase after path reversal"
         assert frech_matrix[1,2] >= frech_matrix[0,1], err_msg
 
+    def test_get_num_paths(self, psa):
+        assert psa.get_num_paths() == 3
+
+    def test_get_paths(self, psa):
+        paths = psa.get_paths()
+        assert len(paths) == 3
+        assert isinstance(paths, list)
+
+    def test_psa_pairs_ValueError(self, psa):
+        with pytest.raises(ValueError):
+            psa.psa_pairs
+
+    def test_psa_pairs(self, psa):
+        psa.run_pairs_analysis()
+        assert len(psa.psa_pairs) == 3
+
+    def test_hausdorff_pairs_ValueError(self, psa):
+        with pytest.raises(ValueError):
+            psa.hausdorff_pairs
+
+    def test_hausdorff_pairs(self, psa):
+        psa.run_pairs_analysis(hausdorff_pairs=True)
+        assert len(psa.hausdorff_pairs) == 3
+
+    def test_nearest_neighbors_ValueError(self, psa):
+        with pytest.raises(ValueError):
+            psa.nearest_neighbors
+
+    def test_nearest_neighbors(self, psa):
+        psa.run_pairs_analysis(neighbors=True)
+        assert len(psa.nearest_neighbors) == 3
+
+    @pytest.mark.xfail
```

Comment:
I think I marked it as failure because I don't know why they fail... this needs to be investigated.

Reply:
@sseyler do you have an idea why this test might be failing?

Reply:
@orbeckst Sorry, I haven't been checking my GitHub messages very often since I've moved to BitBucket. What's the status of the test?

Reply:
Nothing changed since I posted the message. I'd like you to look at the code https://github.com/MDAnalysis/mdanalysis/pull/2049/files#diff-d7b0118128e4a5cf1dad1bda8bde2f12R170 and tell me if my assumption that the input paths and the loaded paths should be the same is flawed. Or if not, why would this test fail?

As part of #1745 you wanted to keep these saved paths (https://github.com/MDAnalysis/mdanalysis/issues/1745#issuecomment-352904322), so now we have to test that they are actually written correctly.

Reply:
I just manually ran through the same failed test but had no problem saving and loading the trajectories; both `assert`s passed. I'm not sure what the issue is exactly, but I noticed that the paths that are actually loaded by `PSAnalysis.load()` are actually *not* the fitted paths as stated in the method's docs.

It looks like I need to do several things:
- [ ] Adjust the `PSAnalysis.load()` method to use a keyword to optionally load the original and/or fitted paths. The original paths will probably be what I'll have loaded by default and the user will have an option to load fitted trajectories (unless you're opposed to this for any reason).
- [ ] Adjust the `PSAnalysis.save_paths()` method to use a keyword to optionally save the original and/or fitted paths. The default option should match the default option that will be used for `PSAnalysis.load()`, although it could also just load the original paths by default and then, by default, also attempt to load any fitted paths it finds.
- [ ] Decide whether 
- [ ] Possibly change `PSAnalysis.load()` to `PSAnalysis.load_paths()`. This breaks backwards compatibility, but is consistent with `PSAnalysis.save_paths()`, is clearer about what's actually being loaded, and eliminates ambiguity given that the `PSAnalysis.save()` method is being removed.

Reply:
Thanks for the analysis.

If you want to change the API then _now_ is a good time. First we would add deprecations (0.19.1 would be ideal) and then change and remove in 0.20.0 (or 1.0?)

As a general rule, simple methods/functions that are doing one thing are better than ones that do wildly different ones.

Reply:
The most basic issue with the test's failing was a problem with inconsistent naming conventions in the `Path` and `PSAnalysis` classes for default filenames for writing `Path` coordinates as .npy files. This problem was addressed by dealing with the most pertinent issue, which was the handling of the default (base) filename used for saving `Path` coordinates that are the result of rmsd fitting by `AlignTraj`. The semi-permanent solution I came up with was to define two default instance variables for the `PSAnalysis` class:

```python
self.default_fitted_path_prefix = 'psafit_'
self.default_path_basename = 'path'
```

specifying the prefix and filename to default to when saving trajectories ("DCD," "TRR," etc.) and the (`numpy`) coordinates of `Path` objects ("npz," "npy"). The user can still specify a "prefix" and a "postfix" to the default filename (or choose their own filename, to which the prefix will be prepended and the postfix will be appended) in the `PSAnalysis.generate_paths()` method, which will then be passed to `Path.run()` which in turn calls `Path.fit_to_reference()`. `Path.fit_to_reference()` appends the user's selected postfix to the filename to get an updated filename, but leaves the prefix separately specified. Then, `Path.fit_to_reference()` constructs an `AlignTraj` object, passing `prefix` and `filename` as parameters to the constructor; by default, `prefix` is set to "psafit_" as a default value in `Path.fit_to_reference()` in order to be consistent with the default prefix used by `AlignTraj` ("rmsfit_"). The point of leaving "psa" in the prefix is to differentiate the source of the need for rmsd fitting from PSA from other possibilities. Also, `PSAnalysis.generate_paths()` method appends a 3-digit number to each trajectory filename, after the user-selected postfix is appended, to differentiate the (multiple) trajectories (i.e., paths) that were loaded into `PSAnalysis`.

Reply:
@sseyler thanks for the explanations. Do you have a PR for these changes? You can either 

1. do a PR against this branch or alternatively, 
2. create this branch in your fork, add your code to the branch, and  submit a new PR with my old commits and your new ones. I would then close this PR.

Reply:
@orbeckst I set up a PR (#2239) against my fork of this branch (more-psa-tests).